karma/karma-puppeteer-client/client.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+;(function (global) {
+  /* global karmaPuppeteer */
+  'use strict'
+
+  function noCleanup() {}
+
+  function withSelector(methodName) {
```

Comment:
Can you add some comments for this function? It's not immediately obvious what this does.

Reply:
Will do. But for context: Puppeteer can only do APIs like this: `click('.selector')`. But I *also* want to enable here a simpler form: `click(element)`, b/c most of time you will already have an element available in the test. So, when an element is passed to these methods, I set a temporary unique attribute on it and instead pass the CSS selector to Puppeteer.